pkg/server/v3/delta_test.go

```diff
@@ -457,3 +457,93 @@ func TestDeltaCallbackError(t *testing.T) {
 		})
 	}
 }
+
+// BENCHMARKS =====================================================================================================
+
+func BenchmarkDeltaResponseHandlers(b *testing.B) {
+	for _, typ := range testTypes {
+		b.Run(typ, func(b *testing.B) {
+			config := makeMockConfigWatcher()
+			config.deltaResponses = makeDeltaResponses()
+			s := server.NewServer(context.Background(), config, server.CallbackFuncs{})
+
+			for n := 0; n < b.N; n++ {
+				resp := makeMockDeltaStream(&testing.T{})
+				// make a wildcard request
+				resp.recv <- &discovery.DeltaDiscoveryRequest{Node: node, TypeUrl: typ, ResourceNamesSubscribe: []string{}}
+				go func() {
+					err := process(typ, resp, s)
+					if err != nil {
+						b.Error(err)
+					}
+				}()
+
+				select {
+				case <-resp.sent:
+					close(resp.recv)
+				case <-time.After(1 * time.Second):
+					b.Fatalf("got no response")
+				}
+			}
+		})
+	}
+}
+
+func BenchmarkDeltaAggregatedHandlers(b *testing.B) {
+	config := makeMockConfigWatcher()
+	config.deltaResponses = makeDeltaResponses()
+	s := server.NewServer(context.Background(), config, server.CallbackFuncs{})
+
+	for n := 0; n < b.N; n++ {
+		resp := makeMockDeltaStream(&testing.T{})
+
+		reqs := []*discovery.DeltaDiscoveryRequest{
+			{
+				Node:    node,
+				TypeUrl: rsrc.ListenerType,
+			},
+			{
+				Node:    node,
+				TypeUrl: rsrc.ClusterType,
+			},
+			{
+				Node:                   node,
+				TypeUrl:                rsrc.EndpointType,
+				ResourceNamesSubscribe: []string{clusterName},
+			},
+			{
+				TypeUrl:                rsrc.RouteType,
+				ResourceNamesSubscribe: []string{routeName},
+			},
+			{
+				TypeUrl:                rsrc.SecretType,
+				ResourceNamesSubscribe: []string{secretName},
+			},
+		}
+
+		for _, r := range reqs {
+			resp.recv <- r
+		}
+
+		go func() {
+			err := s.DeltaAggregatedResources(resp)
+			if err != nil {
+				b.Error(err)
+			}
+		}()
+
+		count := 0
+		for {
+			select {
+			case <-resp.sent:
+				count++
+				if count >= len(reqs) {
+					close(resp.recv)
+					return
+				}
+			case <-time.After(1 * time.Second):
+				b.Fatalf("got %d messages on the stream, not 5", count)
```

Comment:
This error message seems wrong?

Reply:
I took it from the same test above for ADS:
```go
	for {
		select {
		case <-resp.sent:
			count++
			if count >= len(reqs) {
				close(resp.recv)
				assert.Equal(
					t,
					map[string]int{rsrc.EndpointType: 1, rsrc.ClusterType: 1, rsrc.RouteType: 1, rsrc.ListenerType: 1, rsrc.SecretType: 1},
					config.deltaCounts,
				)
				return
			}
		case <-time.After(1 * time.Second):
			t.Fatalf("got %d messages on the stream, not 5", count)
		}
```
But I guess we could remove since it's a benchmark. We just removed the assert to save computations